include/scale/detail/fixed_width_integer.hpp

```diff
@@ -72,7 +72,7 @@ namespace scale::detail {
     // and represent it as native-endian unsigned integer
     uint64_t v = 0u;
 
-    for (size_t i = 0; i < size; ++i) {
+    for (size_t i = 0; i < size && stream.hasMore(1); ++i) {
```

Comment:
Insufficient input is an error.
`nextByte` throws if that error happens.
This behavior is expected.

If loop stops on that condition,
function will return success,
and returned value will be invalid.
This behavior is not expected.